src/codeu/chat/client/Controller.java

```diff
@@ -46,13 +48,17 @@ public Message newMessage(Uuid author, Uuid conversation, String body) {
 
     try (final Connection connection = source.connect()) {
 
-      Serializers.INTEGER.write(connection.out(), NetworkCode.NEW_MESSAGE_REQUEST);
-      Uuid.SERIALIZER.write(connection.out(), author);
-      Uuid.SERIALIZER.write(connection.out(), conversation);
-      Serializers.STRING.write(connection.out(), body);
+      PrintWriter out = new PrintWriter(connection.out(), true);
+      BufferedReader in = new BufferedReader(new InputStreamReader(connection.in()));
+      
```

Comment:
One easy way to package objects into single line is through some sort of json serializer.  I'm sure there's a library out there that does this for you somewhere that just serializes all public variables.